hack/examples/pipeline-gradle.yaml

```diff
@@ -53,7 +53,7 @@ spec:
         - fetch-repository
       params:
         - name: MAVEN_MIRROR_URL
-          value: http://jvm-build-workspace-artifact-cache.$(context.taskRun.namespace).svc.cluster.local/v1/cache/default/0/
+          value: http://jvm-build-workspace-artifact-cache.$(context.taskRun.namespace).svc.cluster.local
```

Comment:
I cannot say with certainty, but at least with the artifact cache referencing localstack, this reliable pattern of service level http URLs no longer held up, because the name of the namespace in kcp is mapped to a `kcp-<random id>` namespace on the workload cluster.

There is I'm sure (though I have not had a chance to look for / discover / analyze it yet, so between you and I we will need to sort that out) a solution in kcp for this.

Reply:
so I've talked to a few folks now and am *pretty sure* we'll need to employ ingress and the haproxy stuff like Frederic is doing for the pipeline as code integration in pr 232 in https://github.com/openshift-pipelines/pipeline-service

this new ingress route / haproxy combo should allow a "non kcp aware" client within the initial pipeline run to access the artifact cache deployment our controller created when it sees the userconfig in the same namesapce the initial pipelinerun is in

diving into this is next on my list after getting #270 properly vetted and merged, and then seeing about our integration in infra-deployments